src/filter_parameters.rs

```diff
@@ -19,8 +25,85 @@ pub struct FilterParameters {
     pub drive: ParameterF32,
     pub mode: ParameterUsize,
     pub slope: ParameterUsize,
-    pub filter_type: AtomicUsize,
+    pub filter_type: ParameterUsize,
+
+    pub reserved: ParameterUsize,
+}
+
+#[repr(i32)]
+#[derive(FromPrimitive, Eq, PartialEq, Debug, EnumIter)]
+pub enum FilterParameterNr {
```

Comment:
Maybe rename this to something simpler? Maybe just ParamIndex? Just to make the code a little less wordy